doc/source/reference/arrays.promotion.rst

```diff
@@ -6,90 +6,104 @@
 Data type promotion in NumPy
 ****************************
 
-When mixing two different data types NumPy has to find the correct dtype for
-the result of the operation.  This step is referred *promotion* or finding
-the common dtype.
-In most you do not have to know much about the details of promotion since
-the promotion step generally ensures that the result of:
-1. Combining multiple arrays e.g. with ``np.concatenate``
-2. The evaluation of any mathematical operation
-
-will return in a result that either matches its input dtypes or has a higher
-precision.
-The ``dtype`` of the result can be found using the `result_type` and
-`promote_types` functions for most (but not all) operations.
```

Comment:
These seemed like details that may not belong at the top of the document.

If these promotion rules do not apply to some operations, or if `result_type` or `promote_types` don't accurately predict the result types of some operations, these can be explained in more detail later.